econml/metalearners.py

```diff
@@ -150,39 +149,42 @@ def fit(self, Y, T, X, inference=None):
         self : an instance of self.
         """
         # Check inputs
+        if X is None:
+            X = np.ones((Y.shape[0], 1))
         Y, T, X, _ = check_inputs(Y, T, X, multi_output_T=False)
         T = self._label_encoder.fit_transform(T)
         T = self._one_hot_encoder.fit_transform(T.reshape(-1, 1))
-        self._d_t = T.shape[1:]
-        self._d_y = Y.shape[1:]
+        self._d_t = T.shape[1] - 1
         feat_arr = np.concatenate((X, T), axis=1)
         self.overall_model.fit(feat_arr, Y)
 
-    def const_marginal_effect(self, X):
+    def const_marginal_effect(self, X=None):
         """Calculate the constant marginal treatment effect on a vector of features for each sample.
 
         Parameters
         ----------
-        X : matrix, shape (m × dₓ)
+        X : matrix, shape (m × dₓ), optional
             Matrix of features for each sample.
 
         Returns
         -------
-        τ_hat : matrix, shape (m, d_y, n_t-1), n_t is number of treatments
+        τ_hat : matrix, shape (m, d_y, d_t)
             Constant marginal CATE of each treatment on each outcome for each sample X[i].
             Note that when Y is a vector rather than a 2-dimensional array,
             the corresponding singleton dimensions in the output will be collapsed
         """
         # Check inputs
+        if X is None:
+            X = np.ones((1, 1))
```

Comment:
As above, leave empty

```suggestion
            X = np.empty((1, 0))
```